### PR TITLE
Offline members after signalling key shouldn't be in group

### PR DESCRIPTION
**Pull request by nikkolasg**

### CODE REVIEW

File: core/util_test.go [hunk 1]
```diff
@@ -347,6 +351,7 @@ func (d *DrandTest2) RunReshare(oldRun, newRun, newThr int, timeout time.Duratio
 		require.NoError(d.t, err)
 		_, err = client.InitReshare(leader.drand.priv.Public, secret, d.groupPath, force)
 		if err != nil {
+			fmt.Println("error in NON LEADER: ", err)
```
Comment (willscott) [inline core/util_test.go:354]: printlns should be turned into logs or removed before merge :)

**Review by willscott — COMMENTED** [review]
There are 3 choices for what happens when a node fails in the reshare:
1 (current): there's a case the group transitions to a new group including the failed node. the failed node never saves it's share, so the resulting group is deficient.
2 (what i believe this change moves us to): the group transitions to a new group not including the failed node. while less members, threshold remains at it's initial setting.
3: the reshare cancels, and the original group is left intact.
(downside is that one misbehaving node in the incoming group can prevent reshares, but if that node can be identified and not added out of band, it's not a permanent DOS)

there's a case to be made that option 3 is what we'd most like to have happen.

File: core/util_test.go (hunk 1 above)
Comment (nikkolasg) [reply core/util_test.go:354]: wait until I mark it as ready before reviewing !! :D

**Comment by jsoares** [comment]
From a network management perspective, I also have a preference for (3). Nevertheless, (2) is better than (1) as it makes recovery easier.

**Comment by nikkolasg** [comment]
Also while thinking about it, I'm not sure we can implement option 3 at all, as that requires a consensus. WHen node A tells to all other nodes "my connection with node B dropped", everybody is supposed to stop the DKG then. But how do we trust node A ;) ? 
And if we rely on "i only stop DKG if my connection as well dropped with node B" then we run into potential weaknesses/bugs, where some node see their connection down and some not. This kind of stuff requires consensus.
This is why the DKG is "threshold resistant": it allows some nodes to misbehave AND to be offline as well, that's the whole point :)

**Comment by nikkolasg** [comment]
Only _before_ leader starts the DKG (by sending the first packet), it can still stops the process, so if the node fails before the leader sends the final group, then it's possible to stop. We can do that it's fine. But as soon as DKG is started, we're not trusting the leader anymore so it's not possible.

**Comment by willscott** [comment]
Appears to be breaking the `TestDrandDKGBroadcastDeny - core` test:
```
ERRROR:  rpc error: code = Unknown desc = drand: err during DKG: drand: error from dkg: node can only process justifications after processing responses
```

**Comment by nikkolasg** [comment]
Mhhh it passes locally :|

**Comment by nikkolasg** [comment]
I've softened a bit the parameters of the test for Travis

**Review by willscott — APPROVED** [review]

**Pull request merged**